### PR TITLE
Fix compiler errors

### PR DESCRIPTION
**Pull request by tmarrinan**
Added catch for is_base_of being false. Also needed to change data type to a pointer rather than struct itself.
Please double check that the false case should do nothing - I just created a dummy method that does nothing so that it could compile.

### CODE REVIEW

File: CMake/Collage.spec [hunk 1]
```diff
@@ -1,13 +1,13 @@
...
+Version:	
 Release:	1%{?dist}
 Summary:	Cross-platform C++ network library
 
 Group:		Development/Libraries
 License:	LGPLv2
 URL:		http://www.libcollage.net/
 Source0:	http://www.equalizergraphics.com/downloads/%{name}-%{version}.tar.gz
-Patch0:		Collage-1.2.1-build-fix.patch
+Patch0:		Collage--build-fix.patch
```
Comment (eile) [inline CMake/Collage.spec]: Please revert this file

File: co/dataIStream.h [hunk 2]
```diff
@@ -217,12 +217,17 @@ class DataIStream
...
     }
 
     /** Read a serializable object. */
     template <class T>
-    void _readSerializable(T& value, const boost::true_type&);
+    void _readSerializable(T& value, const boost::true_type*);
+
+    /** Catch for non-serializable object. */
+    template <class T>
+    void _readSerializable(T&, const boost::false_type*);
```
Comment (eile) [inline co/dataIStream.h:230]: Why is this needed? Compilation should fail if a non-Serializable is read, no?

Comment (tmarrinan) [reply co/dataIStream.h:230]: Compilation was failing - this is why I added it. Perhaps that is indicative of an issue elsewhere in the code if this should never happen?

File: CMake/Collage.spec (hunk 1 above)
Comment (tmarrinan) [reply CMake/Collage.spec]: Ok - done. Not sure how the version got lost in the first place, but I added it back.

File: co/dataIStream.h (hunk 2 above)
Comment (eile) [reply co/dataIStream.h:230]: Yes, that's what I'm thinking.

Comment (tmarrinan) [reply co/dataIStream.h:230]: Ok - is this something you would need me to try and track down, or will you be taking a look?

Comment (tmarrinan) [reply co/dataIStream.h:230]: Hi - I'm just following up to see if there is a fix needed on my end, or what the next steps are? Equalizer (which uses Collage) seems like a great option for the work I'm trying to do, but it just won't compile (or perhaps has a different issue with the suggested patch), which means I may unfortunately need to abandon pursuing its use in favor of a different distributed rendering framework

**Comment by sdegrande** [comment]
To be serializable, a type must be trivially_copyable or must derive from servus::Serializable

In my understanding, to be trivially copyable, a type must have a default copy constructor and a default copy assignment.

I added those 2 defaults to several Equalizer types, and now all errors about _read/writeSerializable are gone.
